### PR TITLE
prevent incorrect bundling of hamcrest and awaitility in the HPI

### PR DESCRIPTION
**Pull request by NotMyFault**
https://github.com/jenkinsci/google-compute-engine-plugin/pull/398#issuecomment-1589719240

### CODE REVIEW

**Review by dwnusbaum — COMMENTED** [review]
Looks like `awaitility` should be moved to `test` scope too (not strictly related to Hamcrest, but would be nice to fix it at the same time).

**Review by jtnord — APPROVED** [review]
Thanks @NotMyFault

**Review by dwnusbaum — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -198,6 +198,7 @@
       <groupId>org.awaitility</groupId>
       <artifactId>awaitility</artifactId>
       <version>4.0.3</version>
+      <scope>test</scope>
```
Comment (jglick) [inline pom.xml:201]: see #154

Comment (dwnusbaum) [reply pom.xml:201]: Hmm, see also https://github.com/jenkinsci/google-compute-engine-plugin/issues/200, and https://github.com/GoogleCloudPlatform/gcp-plugin-core-java 0.3.0 still lists `awaitility` as a compile-scope dependency, which has a compile-scope dependency on `hamcrest-core` and `hamcrest-library`, so maybe this needs to be reverted? Also more generally, it looks like the `gcp-client` library being used by this plugin is abandoned, perhaps the plugin should migrate to something in https://github.com/googleapis/google-cloud-java?

**Pull request merged**